src/main/java/org/folio/rest/impl/UploadBatchVoucherExportHelper.java

```diff
@@ -127,7 +129,12 @@ private void updateHolderWithFileFormat(BatchVoucherUploadHolder uploadHolder) {
 
   private Future<Void> updateHolderWithCredentials(BatchVoucherUploadHolder uploadHolder) {
     return batchVoucherExportConfigService.getExportConfigCredentials(uploadHolder.getExportConfig().getId(), requestContext)
-      .onSuccess(uploadHolder::setCredentials)
+      .onSuccess(credentials -> {
```

Comment:
onSuccess is invoked when future already executed, so it can be situation when the caller use result of this feature as succesful result but after some time onSuccess will throw an exception, need to use compose or map here